src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -701,7 +701,7 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
                 if (value is null)
                 {
                     // Clearing SelectedValueBinding makes the SelectedValue the item itself
-                    SelectedValue = SelectedItem;
+                    SetCurrentValue(SelectedValueProperty, SelectedItem);
```

Comment:
I'm also not sure of the implications of changing properties like this with `SetCurrentValue` - it may well be OK, but needs some thought.